evalml/tests/component_tests/test_ordinal_encoder.py

```diff
@@ -0,0 +1,1019 @@
+
+
+def test_ordinal_encoder_output_doubles():
+    X = pd.DataFrame(
+        {
+            "bool": [bool(i % 2) for i in range(100)],
+            "categorical": ["dog"] * 20 + ["cat"] * 40 + ["fish"] * 40,
+            "integers": [i for i in range(100)],
+            "doubles": [i * 1.0 for i in range(100)],
+        },
+    )
+    X.ww.init(
+        logical_types={
+            "categorical": Ordinal(order=["dog", "cat", "fish"]),
+        },
+    )
+    y = pd.Series([i % 2 for i in range(100)])
+    y.ww.init()
+    ordinal_encoder = OrdinalEncoder()
+    output = ordinal_encoder.fit_transform(X, y)
+    for name, types in output.ww.types["Logical Type"].items():
+        if name == "integers":
+            assert str(types) == "Integer"
+        elif name == "bool":
+            assert str(types) == "Boolean"
+        else:
+            assert str(types) == "Double"
+    assert len(output.columns) == len(X.columns)
+
+
+@pytest.mark.parametrize("data_type", ["list", "np", "pd_no_index", "pd_index", "ww"])
+def test_data_types(data_type):
+    if data_type == "list":
+        X = [["a"], ["b"], ["c"]] * 5
+    elif data_type == "np":
+        X = np.array([["a"], ["b"], ["c"]] * 5)
+    elif data_type == "pd_no_index":
+        X = pd.DataFrame(["a", "b", "c"] * 5)
+    elif data_type == "pd_index":
+        # --> doing int 0 here might defeat the purpose of the no index one?
+        X = pd.DataFrame(["a", "b", "c"] * 5, columns=[0])
+    elif data_type == "ww":
+        X = pd.DataFrame(["a", "b", "c"] * 5)
+        X.ww.init(
+            logical_types={
+                0: Ordinal(order=["a", "b", "c"]),
+            },
+        )
+    encoder = OrdinalEncoder()
+    encoder.fit(X)
+    X_t = encoder.transform(X)
+
+    if data_type != "ww":
+        # Woodwork wont infer Ordinal, so none of the other types will encode features
+        assert not encoder.features_to_encode
+        expected_df = pd.DataFrame(
+            [["a"], ["b"], ["c"]] * 5,
+            columns=[0],
+            dtype="category",
+        )
+        pd.testing.assert_frame_equal(X_t, expected_df)
+    else:
+        assert list(X_t.columns) == ["0_ordinally_encoded"]
+        expected_df = pd.DataFrame(
+            [[0], [1], [2]] * 5,
+            columns=["0_ordinally_encoded"],
+            dtype="float64",
+        )
+        pd.testing.assert_frame_equal(X_t, expected_df)
+
+
+"""
+Tests I didn't include from the ohe tests and why
+
+were not relevant to the ordinal encoder
+    - test_drop_first
+    - test_drop_binary
+    - test_drop_parameter_is_array
+    - test_drop_binary_and_top_n_2
+    - test_ohe_column_names_unique
+Couldn't understand the reason for
+    - test_categorical_dtype
+    - test_all_numerical_dtype
+    - test_ordinal_encoder_woodwork_custom_overrides_returned_by_components
+Seemed redundant to other tests
+    - test_more_top_n_unique_values_large
+    - test_large_number_of_categories - kind of just another test of top_n arg
```

Comment:
I'll remove this comment, but I wanted to highlight the tests from the one hot encoder that I didn't include for the ordinal encoder. 

The top section here is just for tests that were for behaviors that don't exist on this encoder, but for the bottom two sections, I'm less confident in having not included them and just wanted to see if anyone had particularly strong feelings about me not including them.

Reply:
IMO all your choices seem correct. I think the `test_categorical_dtype`/`test_all_numeric_dtype` is to ensure the component only works on the correct columns (and nothing more). As long as we have a test in this file where none of the columns in the (woodwork-typed) input data are Ordinal, that should be covered for this case.

Reply:
Yep, we're covered with `test_ordinal_encoder_is_no_op_for_df_of_non_ordinal_features`